### PR TITLE
 Move convenience methods to vmutils package

### PR DESCRIPTION
**Pull request by paulmey**
Builds on PR #64 to clean up the client to bare API methods. Convenience methods were cleaned up and moved to vmutils.

LMK if you want to see more/less functionality in vmutils

### CODE REVIEW

**Comment by ahmetalpbalkan** [comment]
can you add some readme examples as well. also we should probably review this after #64 is merged.

**Comment by paulmey** [comment]
I split this up into smaller chunks which should be easier to review

**Comment by paulmey** [comment]
rebased

**Comment by paulmey** [comment]
LMK what you'd like changed before merging

File: management/vmutils/deployment.go [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+)
+
+// Configures vm role to deploy from a remote image source. "remoteImageSourceUrl" can be any publically
+// accessible URL to a VHD file, including but not limited to a SAS Azure Storage blob url. "os" needs to be
+// either "Linux" or "Windows". "label" is optional.
+func ConfigureDeploymentFromRemoteImage(role *Role, remoteImageSourceUrl, os, newDiskName, destinationVhdStorageUrl, label string) error {
+	if role == nil {
+		return fmt.Errorf(errParamNotSpecified, "role")
+	}
+	if remoteImageSourceUrl == "" {
```
Comment (ahmetalpbalkan) [inline management/vmutils/deployment.go]: remove nilchecks

File: management/vmutils/deployment.go [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+	return nil
+}
+
+// Configures vm role to deploy from a platform image. See osimage package for methods to retrieve a list of
+// the available platform images. "label" is optional.
+func ConfigureDeploymentFromPlatformImage(role *Role, imageName, destinationVhdStorageUrl, label string) error {
+	if role == nil {
+		return fmt.Errorf(errParamNotSpecified, "role")
+	}
+	if imageName == "" {
```
Comment (ahmetalpbalkan) [inline management/vmutils/deployment.go]: nil checks redundant

**Comment by ahmetalpbalkan** [comment]
LGTM

**Comment by paulmey** [comment]
Thanks for all your time and effort!

**Comment by waitingkuo** [comment]
Thanks~ this is nice!

**Pull request merged**